src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesDataRepairResource.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.casenotes
+
+import com.microsoft.applicationinsights.TelemetryClient
+import io.swagger.v3.oas.annotations.Operation
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.trackEvent
+
+@RestController
+class CaseNotesDataRepairResource(
+  private val caseNotesSynchronisationService: CaseNotesSynchronisationService,
+  private val telemetryClient: TelemetryClient,
+) {
+  @DeleteMapping("/casenotes/{nomisCaseNoteId}/repair")
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ROLE_MIGRATE_CASENOTES')")
```

Comment:
I normally prefer `PreAuthorize` to be at `class` level unless needed lower down.  Prevents new functions being added that aren't protected by accident.